torch/distributed/_sharded_tensor/api.py

```diff
@@ -271,13 +271,11 @@ def gather(
         # will revise this part with CPU support and use dist.gather()
         # once NCCL support for gather() is ready
         # https://github.com/pytorch/pytorch/issues/66187
-        device = torch.device(f"cuda:{rank % world_size}")
-        with torch.cuda.device(device):
-            dist.all_gather_object(
-                obj=local_shards,
-                object_list=gathered_shards,
-                group=self._process_group,
-            )
+        dist.all_gather_object(
```

Comment:
Can we also see if we can remove this one too: https://github.com/pytorch/pytorch/blob/master/torch/distributed/_sharded_tensor/__init__.py#L424